File: fs-tests/projectzipper.fs
module regiondeployer.tests.projectzipper

open regiondeployer.directoryzipper
open Xunit
open regiondeployer.constants.independentconstants
open regiondeployer.randomnamegenerator
open System.Configuration
open Xunit.Abstractions
open regiondeployer.constants.pathconstants
open regiondeployer.constants.personalpaths
open regiondeployer

type public when_it_zips_up_a_project(inputlogger:ITestOutputHelper) =
    do
        configadapter.initialise();

    member private x.logger = inputlogger

    [<Fact>]
    member public this.it_succeeds() : string =
        let zipfilename = createrandomname(defaultzipname, 20)
        ZipUpProject (regionalfarmersroot, zipfilename, this.logger.WriteLine) 

File: tests/ProjectZipper.cs
using Xunit;
using Xunit.Abstractions;
using regiondeployer.directoryzipper;
using regiondeployer.constants.independentconstants;
using regiondeployer.randomnamegenerator;
using System.Configuration;
using regiondeployer.constants.pathconstants;
using regiondeployer.constants.personalpaths;
using regiondeployer;

namespace Tests
{
    public class WhenItZipsUpAProject
    {
        private readonly ITestOutputHelper _logger;

        public WhenItZipsUpAProject(ITestOutputHelper inputLogger)
        {
            configadapter.initialise();
            _logger = inputLogger;
        }

        [Fact]
        public string ItSucceeds()
        {
            string zipFileName = createrandomname(defaultzipname, 20);
            ZipUpProject(regionalfarmersroot, zipFileName, _logger.WriteLine);
            return zipFileName;
        }
    }
}
